DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/src/CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+set_target_properties(${REGISTER_MAP_FUNCTOR_SIMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${SIMULATOR_COMPILE_FLAGS}")
+set_target_properties(${REGISTER_MAP_FUNCTOR_SIMULATOR_TARGET} PROPERTIES LINK_FLAGS "${SIMULATOR_LINK_FLAGS}")
+
+add_executable(${STREAMING_FUNCTOR_SIMULATOR_TARGET} ${STREAMING_FUNCTOR_SOURCE_FILE})
+target_include_directories(${STREAMING_FUNCTOR_SIMULATOR_TARGET} PRIVATE ../../../../../include)
+set_target_properties(${STREAMING_FUNCTOR_SIMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${SIMULATOR_COMPILE_FLAGS}")
+set_target_properties(${STREAMING_FUNCTOR_SIMULATOR_TARGET} PROPERTIES LINK_FLAGS "${SIMULATOR_LINK_FLAGS}")
+
+add_executable(${REGISTER_MAP_LAMBDA_SIMULATOR_TARGET} ${REGISTER_MAP_LAMBDA_SOURCE_FILE})
+target_include_directories(${REGISTER_MAP_LAMBDA_SIMULATOR_TARGET} PRIVATE ../../../../../include)
+set_target_properties(${REGISTER_MAP_LAMBDA_SIMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${SIMULATOR_COMPILE_FLAGS}")
+set_target_properties(${REGISTER_MAP_LAMBDA_SIMULATOR_TARGET} PROPERTIES LINK_FLAGS "${SIMULATOR_LINK_FLAGS}")
+
+add_executable(${STREAMING_LAMBDA_SIMULATOR_TARGET} ${STREAMING_LAMBDA_SOURCE_FILE})
+target_include_directories(${STREAMING_LAMBDA_SIMULATOR_TARGET} PRIVATE ../../../../../include)
+set_target_properties(${STREAMING_LAMBDA_SIMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${SIMULATOR_COMPILE_FLAGS}")
+set_target_properties(${STREAMING_LAMBDA_SIMULATOR_TARGET} PROPERTIES LINK_FLAGS "${SIMULATOR_LINK_FLAGS}")
+
+add_custom_target(fpga_sim DEPENDS ${REGISTER_MAP_FUNCTOR_SIMULATOR_TARGET} ${STREAMING_FUNCTOR_SIMULATOR_TARGET} ${REGISTER_MAP_LAMBDA_SIMULATOR_TARGET} ${STREAMING_LAMBDA_SIMULATOR_TARGET})
+
+###############################################################################
+### Generate Report
+###############################################################################
+# To compile manually:
+#   icpx -fsycl -fintelfpga -Xssimulation -Xsboard=<FPGA_BOARD> -fsycl-link=early register_map_functor_model.cpp -o register_map_functor_model.a
+#   icpx -fsycl -fintelfpga -Xssimulation -Xsboard=<FPGA_BOARD> -fsycl-link=early streaming_functor_model.cpp -o streaming_functor_model.a
+#   icpx -fsycl -fintelfpga -Xssimulation -Xsboard=<FPGA_BOARD> -fsycl-link=early register_map_lambda_model.cpp -o register_map_lambda_model.a
+#   icpx -fsycl -fintelfpga -Xssimulation -Xsboard=<FPGA_BOARD> -fsycl-link=early streaming_lambda_model.cpp -o streaming_lambda_model.a
+# The compile output is not an executable, but an intermediate compilation result unique to SYCL.
+set(REGISTER_MAP_FUNCTOR_FPGA_EARLY_IMAGE ${REGISTER_MAP_FUNCTOR_TARGET_NAME}_report.a)
+add_executable(${REGISTER_MAP_FUNCTOR_FPGA_EARLY_IMAGE} ${REGISTER_MAP_FUNCTOR_SOURCE_FILE})
+target_include_directories(${REGISTER_MAP_FUNCTOR_FPGA_EARLY_IMAGE} PRIVATE ../../../../../include)
+set_target_properties(${REGISTER_MAP_FUNCTOR_FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${SIMULATOR_COMPILE_FLAGS}")
```

Comment:
Is this not generating a warning about -Xssimulation not being used?
Note that now your regtest can (and should) be run with the "report" flow as well

Reply:
switched to use the hardware flow for reports